docs/element-definitions.md

```diff
@@ -1,1419 +1,1623 @@
-          "element": "array",
-          "content": [
-            {
-              "element": "string",
-              "content": "resourceGroup"
-            }
-          ]
-        },
-        "title": {
-          "element": "string",
-          "content": "Question"
-        }
-      },
-      "content": [
-        {
-          "element": "copy",
-          "content": "Resources related to questions in the API."
-        }
-      ]
+      "element": "string"
+    },
+    {
+      "element": "number"
     }
   ]
 }
+
 ```
 
-### Copy (Base API Element)
+---
 
-Copy element represents a copy text—a textual information in API description.
-Its content is a string and it MAY include information about the media type
-of the copy's content.
+### Member Element
+Type with domain of all _properties_.
 
-Unless specified otherwise, a copy element's content represents the
-description of its parent element and SHOULD be used instead of parent
-element's description metadata.
+Quoting from the [Element section](#element):
+> A _property_ is a tuple (`key`, `value`) where
+> - `key` SHALL be a non-empty, finite character string
+> - `value` SHALL be an _Element_
+> - two properties SHALL be equal if their keys are.
 
-#### Properties
+#### Template
+- `element` - `"member"`
+- `attributes`
+  - `typeAttributes` ([Array](#array-element)[[String](#string-element)])
+    - `required` ([String](#string-element)) - Property MUST be present in Object containing it. I.e. restricts the domain of the containing Object Element type to one containing this property.
+    - `optional` ([String](#string-element)) - Property MAY NOT be present in Object containing it. I.e. expands the domain of the containing Object Element type to one not containing this property.</p>
+  - `variable` - ([Boolean](#boolean-element)) - Property key SHALL be interpreted as a variable name instead of a literal name
+  - `validation` - _reserved for future use_
+- `content`
+  - `key` - an Element representing a key; SHOULD be a [String Element](#string-element)
+  - `value` - an Element representing the value
 
-- `element`: copy (string, fixed)
+#### Examples
+See [Object Element](#object-element) for examples.
+
+---
+
+### Object Element
+
+Type with domain of all finite lists of _properties_.
+
+Quoting from the [Element section](#element):
+> A _property_ is a tuple (`key`, `value`) where
+> - `key` SHALL be a non-empty, finite character string
+> - `value` SHALL be an _Element_
+> - two properties SHALL be equal if their keys are.
+
+#### Template
+- `element` - `"object"`
 - `attributes`
-    - `contentType` ([String Element][]) - Optional media type of the content.
-- `content` (string)
+  - `typeAttributes` ([Array](#array-element)[[String](#string-element)])
+    - `fixed` ([String](#string-element)) - SHALL restrict the domain to a specific instance of a property list. Entries in `content` of this Object Element MUST contain at least one of content, default and sample.
```

Comment:
Restricts the domain to a fixed-length unordered list of properties, making them implicitly required. It further applies fixed to an item given it declares content or default.